### ISSUE TITLE
CMake variable LLVM_DIR might not be used if another LLVM install is present

### ISSUE THREAD

**Issue by Aympab** [AdaptiveCpp/AdaptiveCpp#991]
Hi, I ran into a small problem while trying to compile OpenSYCL with a specific LLVM installation using `LLVM_DIR` cmake variable.
When reading the LLVM [dependency installation instructions](https://github.com/OpenSYCL/OpenSYCL/blob/develop/doc/install-llvm.md#pointing-open-sycl-to-the-right-llvm), we might think (at least I did) that setting the `LLVM_DIR` variable is sufficient, but actually, if another LLVM installation is available on the computer (and environment variables are set to this install), it will ignore the `LLVM_DIR` because the CMake script does `find_package` before using LLVM_DIR :

https://github.com/OpenSYCL/OpenSYCL/blob/2b0067b58780663a8ea651bcc7ef24a7ae248ff7/CMakeLists.txt#L133

The workaround is to specify `CMAKE_PREFIX_PATH` to the `llvm/lib/cmake` directory. But I think it might be confusing only reading the documentation.

It's just some details but I think some people might run into the same problem as I did

**Comment by illuhad** [comment]
`LLVM_DIR` is not a variable introduced by Open SYCL, it comes from LLVM's cmake integration. Shouldn't `find_package()` already take `LLVM_DIR` into account?

**Comment by Aympab** [comment]
Yes, it does. I meant that reading the OpenSYCL doc install for LLVM can be confusing because it says to set the `LLVM_DIR` variable toward LLVM's cmake path. But if the `find_package(...)` finds a system LLVM, the CMake script will not take into account the user-defined `LLVM_DIR` from CMake's command line. This is why I had to manually set `CMAKE_PREFIX_PATH`
Not sure if I'm clear. And as I said, it is details.

**Comment by Aympab** [comment]
Ok I just checked more in-depth and actually this is my mistake.
Actually the `LLVM_DIR` has to point towards a directory containing the `LLVM-Config.cmake` file. When building LLVM from source, this file is located inside `$LLVM_INSTALL_DIR/lib/cmake/llvm`, the last directory being `llvm`. The Open SYCL [documentation for LLVM dependency ](https://github.com/OpenSYCL/OpenSYCL/blob/develop/doc/install-llvm.md#pointing-open-sycl-to-the-right-llvm) says : "`-DLLVM_DIR=/path/to/llvm/cmake` must be pointed to your LLVM installation, specifically, the subdirectory containing the LLVM cmake files.". I got tricked because the example path ends with the `cmake` folder, so I was pointing my `LLVM_DIR` variable towards `$LLVM_INSTALL_DIR/lib/cmake` rather than `$LLVM_INSTALL_DIR/lib/cmake/llvm`.

When specifying the `CMAKE_PREFIX_PATH`, cmake will look for the `LLVM-Config.cmake` recursively within the path provided, this is why it was working for me. I got confused by the example path.
